### ISSUE TITLE
libopencv_imgcodecs.so.3.1: error adding symbols: DSO missing from command line (build with anaconda)

### ISSUE THREAD

**Issue by Earthson** [dmlc/mxnet#1333]
I'm trying to build mxnet with Anaconda (gcc-4.8.5, opencv3), here is my error log:

```
. . .
a - build/src/operator/crop_gpu.o
a - build/src/operator/identity_attach_KL_sparse_reg_gpu.o
a - build/src/operator/activation_gpu.o
a - build/src/operator/concat_gpu.o
a - build/src/operator/native_op_gpu.o
a - build/src/operator/cudnn_batch_norm_gpu.o
a - build/src/operator/regression_output_gpu.o
a - build/src/operator/upsampling_gpu.o
a - build/src/operator/swapaxis_gpu.o
a - build/src/operator/cast_gpu.o
a - build/src/operator/convolution_gpu.o
a - build/src/optimizer/sgd_gpu.o
/usr/bin/ld: /tmp/ccps12Mn.o: undefined reference to symbol '_ZN2cv8imencodeERKNS_6StringERKNS_11_InputArrayERSt6vectorIhSaIhEERKS6_IiSaIiEE'
/home/earthson/env/opt/anaconda/lib/libopencv_imgcodecs.so.3.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
Makefile:187: recipe for target 'bin/im2rec' failed
make: *** [bin/im2rec] Error 1
```

`make/config.mk`

```
# the additional link flags you want to add
ADD_LDFLAGS = "-L/home/earthson/env/opt/anaconda/lib"

# the additional compile flags you want to add
ADD_CFLAGS = "-I/home/earthson/env/opt/anaconda/include"

#---------------------------------------------
# matrix computation libraries for CPU/GPU
#---------------------------------------------

# whether use CUDA during compile
USE_CUDA = 1

# add the path to CUDA library to link and compile flag
# if you have already add them to environment variable, leave it as NONE
USE_CUDA_PATH = /home/earthson/env/opt/cuda
#USE_CUDA_PATH = NONE

# whether use CuDNN R3 library
USE_CUDNN = 1

# whether use cuda runtime compiling for writing kernels in native language (i.e. Python)
USE_NVRTC = 1

# whether use opencv during compilation
# you can disable it, however, you will not able to use
# imbin iterator
USE_OPENCV = 1

# use openmp for parallelization
USE_OPENMP = 1
```

**Comment by Earthson** [comment]
with `export PKG_CONFIG_PATH=/home/earthson/env/opt/anaconda/lib/pkgconfig` I got:

```
a - build/src/operator/convolution_gpu.o                                                                                                                                                                                                                                [0/1971]
a - build/src/optimizer/sgd_gpu.o
/usr/bin/ld: cannot find -lippicv
collect2: error: ld returned 1 exit status
Makefile:174: recipe for target 'lib/libmxnet.so' failed
make: *** [lib/libmxnet.so] Error 1
make: *** Waiting for unfinished jobs....
/usr/bin/ld: cannot find -lippicv
collect2: error: ld returned 1 exit status
Makefile:187: recipe for target 'bin/im2rec' failed
make: *** [bin/im2rec] Error 1
```

**Comment by piiswrong** [comment]
Why is your opencv libs under anaconda/lib? How did you install your opencv?

**Comment by Earthson** [comment]
@piiswrong opencv and gcc are installed using conda

```
conda install gcc
conda install opencv3 #using menpo channel
```

I've downloaded the ippicv, the error has gone, But when I am trying to use openblas(also integrated in Anaconda), it shows that libfortran not found.

Here is the error case I've found:

- `opencv3 + openblas`: `libgfortran.so not found`
- `openblas: no error
- `opencv3 + mkl`: no error

- - -

My system is Ubuntu 15.10 with gcc 5.2.

I've also tried to build mxnet using system gcc, there is no compile error, but when I using Anaconda's IPython, an link error occured.
